src/d/actor/d_a_leaflift.cpp

```diff
@@ -4,98 +4,352 @@
+    m43F = 0x1e;
+    mpModel->calc();
+    setMoveBGMtx();
+    dComIfG_Bgsp()->Regist(mcBgW, this);
+    set_mtx();
+    mcBgW->Move();
 }
 
 /* 00000634-00000760       .text _create__9daLlift_cFv */
 s32 daLlift_c::_create() {
-    /* Nonmatching */
+    int res;
+    fopAcM_SetupActor(this, daLlift_c);
+    res = dComIfG_resLoad(&mPhs, m_arcname);
+    if (res == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0xf40)) {
+            return cPhs_ERROR_e;
+        }
+        else {
+            CreateInit();
+        }
+    }
+    return res;
 }
 
 /* 00000918-000009C4       .text nodeCallBack__FP7J3DNodei */
-static BOOL nodeCallBack(J3DNode*, int) {
-    /* Nonmatching */
+static BOOL nodeCallBack(J3DNode* node, int idx) {
+    if (!idx) {
+        J3DJoint* joint = (J3DJoint*)node;
+        s32 jntNo = joint->getJntNo();
+        J3DModel* model = j3dSys.getModel();
+        daLlift_c* i_this = (daLlift_c*)model->getUserArea();
+        if (i_this) {
+            mDoMtx_stack_c::copy(model->getAnmMtx(jntNo));
+            mDoMtx_stack_c::quatM(&i_this->mBaseQuat);
+            cMtx_copy(mDoMtx_stack_c::get(), J3DSys::mCurrentMtx);
+            model->setAnmMtx(jntNo, mDoMtx_stack_c::get());
+            cMtx_copy(mDoMtx_stack_c::get(), i_this->mBGMtx);
+        }
+    }
+    return TRUE;
 }
 
 /* 000009C4-00000CAC       .text rideCallBack__FP4dBgWP10fopAc_ac_cP10fopAc_ac_c */
-void rideCallBack(dBgW*, fopAc_ac_c*, fopAc_ac_c*) {
-    /* Nonmatching */
+static void rideCallBack(dBgW* param1, fopAc_ac_c* i_this, fopAc_ac_c* i_other) {
+    f32 fVar1;
+    f32 fVar2;
+    daLlift_c* param2 = (daLlift_c*)i_this;
+    daLlift_c* param3 = (daLlift_c*)i_other;
+
+    cXyz posOffset = param3->current.pos - param2->current.pos;
+    
+    fVar1 = 2.0f;
+    if (fopAcM_GetName(param3) == 0xa9) {
+        param2->m43C = 1;
+        param2->m43E = 1;
+        posOffset = posOffset.outprod(up_vec);
+        mDoMtx_stack_c::YrotS(-param2->current.angle.y);
+        mDoMtx_stack_c::multVec(&posOffset, &posOffset);
+        if (param2->m43D) {
+            posOffset.x = param3->current.pos.x;
+            posOffset.y = param3->current.pos.y;
+            posOffset.z = param3->current.pos.z;
+            posOffset.x += 200.0f;
+            posOffset = posOffset.outprod(up_vec);
+            mDoMtx_stack_c::YrotS(-param2->current.angle.y);
+            mDoMtx_stack_c::multVec(&posOffset, &posOffset);
+            fVar1 = cM_ssin((param2->m444 & 0x1F) << 11) * 2.0f * 0.25f * (cM_rndFX(0.2f) + 1.0f);
+        }
+        fVar2 = posOffset.abs();
+        if (!posOffset.normalizeRS() ) {
+          return;
+        }
+        cLib_addCalcAngleS2(&param2->m440, -fVar2 * fVar1, 8, 0x200);
+        fVar1 = cM_ssin(param2->m440);
+        param2->mZeroQuat.x = posOffset.x * fVar1;
+        param2->mZeroQuat.y = posOffset.y * fVar1;
+        param2->mZeroQuat.z = posOffset.z * fVar1;
+        param2->mZeroQuat.w = cM_scos(param2->m440);
```

Comment:
`mZeroQuat` is a misleading name for this member. It's initialized to zero, but here it's being set to a nonzero value.